src/approaches/tamp_approach.py

```diff
@@ -61,3 +52,38 @@ def _get_current_predicates(self) -> Set[Predicate]:
         Defaults to initial predicates.
         """
         return self._initial_predicates
+
+
+def option_plan_to_policy(plan: Sequence[_Option]
+                          ) -> Callable[[State], Action]:
+    """Create a policy that executes the options in order.
+
+    We may want to move this out of here later, but I'm leaving
```

Comment:
Let's do this correctly now! We should move this method to utils.py.

We can define a new exception in utils.py, e.g. OptionPlanToPolicyException, and then catch that within TAMPApproach and raise an ApproachFailure with the error message copied over.

Reply:
FYI there's no variation in the error messages, so I'm not doing the very last thing you said here